src/driver/communication_interface.cc

```diff
@@ -554,10 +577,19 @@ void CommunicationInterface::HandlePlan(
     // Cartesian goal command. Not implemented yet. Ignore for now
   }
   lock.unlock();
+
+  auto t_end {hr_clock::now()};
+  timepoints_.t_received = t_start;
+  timepoints_.t_accepted = t_end;
+
```

Comment:
What happens if we receive a new plan while still executing the old one?

Reply:
I remember seeing log spam complaining that "Franka already has a plan".  There must be protection somewhere.

Reply:
There's `{}:CancelPlans: {} still has a plan` but that's in TLE

Reply:
In `JointPositionCallback` we have:
```c++
  if (comm_interface_->HasNewPlan()) {  // pop the new plan and set it up
    auto [new_plan, new_plan_utime, new_plan_exec_opt,
          new_plan_contact_expected] {comm_interface_->PopNewPlan()};

    bool has_active_plan {plan_ != nullptr};
    bool is_new_plan_valid {!new_plan->empty()
                            && new_plan->end_time() > franka_time_};
    if (is_new_plan_valid
        && (!has_active_plan || IsContinuousWithCurrentPlan(new_plan))) {
      // If the new plan is coninuous in position, velocity and acceleration
      // with the current plan at current frank time, replace it with the new
      // plan. Or if we do not currently have a plan, move the new plan's
      // ownership to the current plan unique_ptr
      UpdateActivePlan(std::move(new_plan), new_plan_utime, new_plan_exec_opt,
                       new_plan_contact_expected);
      // the current (desired) position of franka is the starting position:
      start_conf_franka_ = current_conf_franka;

      // print the appropriate warning, send plan complete with success=false
      // and skip this plan
    } else {
      if (!is_new_plan_valid) {
        dexai::log()->warn(
            "JointPositionCallback: new plan with utime: {} is invalid.\n"
            "franka_time: {}\tplan.end_time: {}",
            new_plan_utime, franka_time_, new_plan->end_time());
        comm_interface_->SetPlanCompletion(
            new_plan_utime, false, "discarded because plan was invalid");
      } else {
        dexai::log()->warn(
            "JointPositionCallback: new plan with utime: {} is not continuous "
            "with current plan with utime: {} at t={}",
            new_plan_utime, plan_utime_, franka_time_);
        comm_interface_->SetPlanCompletion(
            new_plan_utime, false,
            fmt::format("discarded because plan is not continuous with current "
                        "plan utime: {} at t={}",
                        plan_utime_, franka_time_));
      }
    }
```

Reply:
Updated the logic to handle this case